File: src/components/companyHeader.ts
import { Company } from '../data.js';

/**
 * Creates and returns an HTMLElement representing a company header.
 * @param {Company} company - The company data.
 * @returns {HTMLElement} The created company header element.
 */
export function createCompanyHeader(company: Company): HTMLElement {
    const companyHeader = document.createElement('div');
    companyHeader.classList.add('company-header');
    companyHeader.innerHTML = `
        <h3>${company.companyName}</h3>
        <p><strong>${company.startDate} - ${company.endDate}</strong></p>
    `;
    return companyHeader;
}

File: src/components/company_header.rs
use crate::data::Company;
use yew::prelude::*;

#[derive(Properties, Clone, PartialEq)]
pub struct Props {
    pub company: Company,
}

#[function_component(CompanyHeader)]
pub fn company_header(props: &Props) -> Html {
    let company = &props.company;
    html! {
        <div class="company-header">
            <h3>{ company.company_name }</h3>
            <p><strong>{ format!("{} - {}", company.start_date, company.end_date) }</strong></p>
            <hr />
        </div>
    }
}
